app/src/main/java/hr/dice/filipbionda/tmdbpractice/ui/welcomescreen/WelcomeScreen.kt

```diff
@@ -0,0 +1,163 @@
+package hr.dice.filipbionda.tmdbpractice.ui.welcomescreen
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import hr.dice.filipbionda.tmdbpractice.R
+import hr.dice.filipbionda.tmdbpractice.ui.theme.TMDBPracticeTheme
+
+@Composable
+fun WelcomeScreen(modifier: Modifier = Modifier) {
+    Box(
+        modifier = modifier.fillMaxSize(),
+    ) {
+        BackgroundImage()
+        WelcomeScreenContent(modifier = Modifier.align(alignment = Alignment.BottomCenter))
+    }
+}
+
+@Composable
+private fun BackgroundImage(modifier: Modifier = Modifier) {
+    val brush =
+        Brush.verticalGradient(
+            colorStops =
+            arrayOf(
+                0.0f to MaterialTheme.colorScheme.onBackground,
+                0.8f to MaterialTheme.colorScheme.background,
+            ),
+        )
+    Box(
+        modifier = modifier.fillMaxSize()
```

Comment:
staticAnalysis fails here because of the missing trailing comma. You can configure in Android Studio to add trailing commas when you format the code.

Reply:
The composable itself in this scenario shouldn't worry about how much space it takes. It would be better to pass this `modifier.fillMaxSize()` in the modifier porperty when calling the `BackgroundImage` composable so it would look something like this

```
@Composable
WelcomeScreen() {
....
        BackgroundImage(Modifier.fillMaxSize())
....
}
....
@Composable
BackgroundImage(modifier: Modifier = Modifier) {
....
Box(modifier = modifier)
}
```

Reply:
In the Composable functions that are acting as components, the caller should worry about components size and placement. That is why it is recommended to always have a modifier parameter and to add it to the first composable component in that component

Reply:
Done. I get it now.